app/controllers/application_controller.rb

```diff
@@ -1,3 +1,8 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+
+  def favorites
```

Comment:
Very Very clever to add the session in the Application Controller so that it is available for all pages and interactions on the website!!!